<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Apply a S3 filter object to a pmartR S3 object — applyFilt • pmartR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Apply a S3 filter object to a pmartR S3 object — applyFilt"><meta property="og:description" content="This function takes a filter object of class 'cvFilt', 'rmdFilt',
  'moleculeFilt', 'proteomicsFilt', 'imdanovaFilt', 'RNAFilt', 'totalCountFilt', 
  or 'customFilt' and  applies the filter to a dataset of pepData, 
  proData, lipidData, metabData, nmrData or 
  seqData."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Convenience_Functions.html">Convenience Functions</a>
    </li>
    <li>
      <a href="../articles/Data_Normalization.html">Data Normalization</a>
    </li>
    <li>
      <a href="../articles/Data_Object_Creation.html">Data Object Creation</a>
    </li>
    <li>
      <a href="../articles/Filter_Functionality.html">Filter Functions</a>
    </li>
    <li>
      <a href="../articles/Graphing_with_pmartR.html">Graphing with pmartR</a>
    </li>
    <li>
      <a href="../articles/RNAseq_Processing_Workflow.html">RNAseq Data Processing</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Trelliscope_Vignette.html">Using Trelliscope Functions with pmartR</a>
    </li>
    <li>
      <a href="../articles/Typical_Processing_Workflow.html">Typical Processing Workflow</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/pmartR/pmartR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Apply a S3 filter object to a pmartR S3 object</h1>
    <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/HEAD/R/applyFilt.R" class="external-link"><code>R/applyFilt.R</code></a></small>
    <div class="hidden name"><code>applyFilt.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes a filter object of class 'cvFilt', 'rmdFilt',
  'moleculeFilt', 'proteomicsFilt', 'imdanovaFilt', 'RNAFilt', 'totalCountFilt', 
  or 'customFilt' and  applies the filter to a dataset of <code>pepData</code>, 
  <code>proData</code>, <code>lipidData</code>, <code>metabData</code>, <code>nmrData</code> or 
  <code>seqData</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for moleculeFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for totalCountFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, <span class="va">min_count</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for RNAFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, min_nonzero <span class="op">=</span> <span class="cn">NULL</span>, size_library <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for cvFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, cv_threshold <span class="op">=</span> <span class="fl">150</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for rmdFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span></span>
<span>  <span class="va">filter_object</span>,</span>
<span>  <span class="va">omicsData</span>,</span>
<span>  pvalue_threshold <span class="op">=</span> <span class="fl">1e-04</span>,</span>
<span>  min_num_biomolecules <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for proteomicsFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span>, min_num_peps <span class="op">=</span> <span class="cn">NULL</span>, redundancy <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for imdanovaFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span></span>
<span>  <span class="va">filter_object</span>,</span>
<span>  <span class="va">omicsData</span>,</span>
<span>  comparisons <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_nonmiss_anova <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min_nonmiss_gtest <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  remove_singleton_groups <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for customFilt</span></span>
<span><span class="fu">applyFilt</span><span class="op">(</span><span class="va">filter_object</span>, <span class="va">omicsData</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>filter_object</dt>
<dd><p>an object of the class 'cvFilt', 'proteomicsFilt',
'rmdFilt', 'moleculeFilt', 'imdanovaFilt', 'customFilt', 'RNAFilt', or
'totalCountFilt' created by <code>cv_filter</code>, <code>proteomics_filter</code>,
<code>rmd_filter</code>, <code>molecule_filter</code>, <code>imdanova_filter</code>,
<code>custom_filter</code>, <code>RNA_filter</code>, or <code>total_count_filter</code>,
respectively.</p></dd>


<dt>omicsData</dt>
<dd><p>an object of the class <code>pepData</code>, <code>proData</code>,
<code>lipidData</code>, <code>metabData</code>, <code>nmrData</code>, or <code>seqData</code>
usually created by <code><a href="as.pepData.html">as.pepData</a></code>, <code><a href="as.proData.html">as.proData</a></code>,
<code><a href="as.lipidData.html">as.lipidData</a></code>, <code><a href="as.metabData.html">as.metabData</a></code>,
<code><a href="as.nmrData.html">as.nmrData</a></code>, or <code><a href="as.seqData.html">as.seqData</a></code>, respectively.</p></dd>


<dt>...</dt>
<dd><p>further arguments</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An object of the class <code>pepData</code>, <code>proData</code>,</p>
<p></p>
<p><code>lipidData</code>, <code>metabData</code>, <code>nmrData</code>, or <code>seqData</code> with
  specified cname_ids, edata_cnames, and emeta_cnames filtered out of the
  appropriate datasets.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Further arguments can be specified depending on the class of the
  <code>filter_object</code> being applied.</p>
<p>For a <code>filter_object</code> of type 'moleculeFilt':</p><table class="table table"><tr><td><code>min_num</code></td><td>an integer value specifying the minimum number of times
  each biomolecule must be observed across all samples in order to retain the
  biomolecule. Default value is 2.</td></tr></table><p>For a <code>filter_object</code> of type 'cvFilt':</p><table class="table table"><tr><td><code>cv_threshold</code></td><td>an integer value specifying the maximum
  coefficient of variation (CV) threshold for the biomolecules. Biomolecules
  with CV greater than this threshold will be filtered. Default threshold is
  150.</td></tr></table><p>For a <code>filter_object</code> of type 'rmdFilt':</p><table class="table table"><tr><td><code>pvalue_threshold</code></td><td>numeric value between 0 and 1, specifying the
  p-value below which samples will be removed from the dataset. Default is
  0.001.</td></tr><tr><td><code>min_num_biomolecules</code></td><td>numeric value greater than 10
  (preferably greater than 50) that specifies the minimum number of
  biomolecules that must be present in order to create an rmdFilt object.
  Using values less than 50 is not advised.</td></tr></table><p>For a <code>filter_object</code> of type 'proteomicsFilt' either or both of the
  following can be applied:</p><table class="table table"><tr><td><code>min_num_peps</code></td><td>an
  optional integer value between 1 and the maximum number of peptides that
  map to a protein in omicsData. The value specifies the minimum number of
  peptides that must map to a protein. Any protein with less than
  <code>min_num_peps</code> mapping to it will be removed from the dataset. Default
  value is NULL, meaning that this filter is not applied.</td></tr><tr><td><code>redundancy</code></td><td>logical indicator of whether to filter out
  degenerate/redundant peptides (peptides that map to more than one protein).
  Default value is FALSE.</td></tr></table><p>For a <code>filter_object</code> of type 'imdanovaFilt':</p><table class="table table"><tr><td><code>min_nonmiss_anova</code></td><td>integer value specifying the minimum number
  of non-missing feature values allowed per group for <code>anova_filter</code>.
  Default value is 2.</td></tr><tr><td><code>min_nonmiss_gtest</code></td><td>integer value
  specifying the minimum number of non-missing feature values allowed per
  group for <code>gtest_filter</code>. Default value is 3.</td></tr></table><p>For a <code>filter_object</code> of type 'totalCountFilt':</p><table class="table table"><tr><td><code>min_count</code></td><td>integer value specifying the minimum number of
  biomolecule counts observed across all samples in order for the biomolecule
  to be retained in the dataset. This filter is only applicable for
  <code>seqData</code> objects.</td></tr></table><p>For a <code>filter_object</code> of type 'RNAFilt' either or both of the
  following can be applied:</p><table class="table table"><tr><td><code>min_nonzero</code></td><td>integer
  value specifying the minimum number of non-zero feature values per sample.</td></tr><tr><td><code>size_library</code></td><td>integer value or fraction between 0 and 1
  specifying the minimum number of total reads per sample. This filter is
  only applicable for <code>seqData</code> objects.</td></tr></table><p>There are no further arguments for a <code>filter_object</code> of type '
  customFilt'.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="molecule_filter.html">molecule_filter</a></code></p>
<p><code><a href="imdanova_filter.html">imdanova_filter</a></code></p>
<p><code><a href="rmd_filter.html">rmd_filter</a></code></p>
<p><code><a href="cv_filter.html">cv_filter</a></code></p>
<p><code><a href="proteomics_filter.html">proteomics_filter</a></code></p>
<p><code><a href="custom_filter.html">custom_filter</a></code></p>
<p><code><a href="RNA_filter.html">RNA_filter</a></code></p>
<p><code><a href="total_count_filter.html">total_count_filter</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Lisa Bramer, Kelly Stratton</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmartR/pmartRdata" class="external-link">pmartRdata</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">to_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pep_object</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">to_filter</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Summary of Molecule Filter </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------------------------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Minimum Number:2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Filtered:930</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Not Filtered:20399</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  num_observations frequency_counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 1              930</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 2             1928</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 3             2895</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 4             3826</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 5             4705</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 6             5627</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 7             6443</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 8             7213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 9             7937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                10             8647</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                11             9383</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                12            10032</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                13            10678</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                14            11292</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                15            11937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                16            12592</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                17            13240</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                18            13917</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                19            14571</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                20            15248</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                21            15941</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                22            16800</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                23            17956</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                24            21329</span>
<span class="r-in"><span><span class="va">pep_object2</span> <span class="op">&lt;-</span> <span class="fu">applyFilt</span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">to_filter</span>,</span></span>
<span class="r-in"><span>                         omicsData <span class="op">=</span> <span class="va">pep_object</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">pep_object2</span><span class="op">)</span> <span class="co"># number of Peptides is as expected based on summary of the filter object that was applied</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Class                         pepData</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Unique SampleIDs (f_data)          24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Unique Peptides (e_data)        20399</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Unique RazorProteins (e_meta)    2916</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Missing Observations           202348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Proportion Missing              0.413</span>
<span class="r-in"><span><span class="va">pep_object2</span> <span class="op">&lt;-</span> <span class="fu"><a href="group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pep_object2</span>, main_effects <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">to_filter2</span> <span class="op">&lt;-</span> <span class="fu"><a href="imdanova_filter.html">imdanova_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pep_object2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pep_object3</span> <span class="op">&lt;-</span> <span class="fu">applyFilt</span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">to_filter2</span>,</span></span>
<span class="r-in"><span>                         omicsData <span class="op">=</span> <span class="va">pep_object2</span>,</span></span>
<span class="r-in"><span>                         min_nonmiss_anova <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> You have specified remove_singleton_groups = TRUE, but there are no singleton groups to remove. Proceeding with application of the IMD-ANOVA filter.</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Lisa Bramer, Kelly Stratton.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

