<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Reduce Dimension of Data for Exploratory Data Analysis — dim_reduction • pmartR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Reduce Dimension of Data for Exploratory Data Analysis — dim_reduction"><meta property="og:description" content="For data types other than seqData, this function calculates principal
components using projection pursuit estimation, which implements an
expectation-maximization (EM) estimation algorithm when data is missing. For
seqData counts, a generalized version of principal components analysis for
non-normally distributed data is calculated under the assumption of a
negative binomial distribution with global dispersion."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Convenience_Functions.html">Convenience Functions</a>
    </li>
    <li>
      <a href="../articles/Data_Normalization.html">Data Normalization</a>
    </li>
    <li>
      <a href="../articles/Data_Object_Creation.html">Data Object Creation</a>
    </li>
    <li>
      <a href="../articles/Filter_Functionality.html">Filter Functions</a>
    </li>
    <li>
      <a href="../articles/Graphing_with_pmartR.html">Graphing with pmartR</a>
    </li>
    <li>
      <a href="../articles/RNAseq_Processing_Workflow.html">RNAseq Data Processing</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Trelliscope_Vignette.html">Using Trelliscope Functions with pmartR</a>
    </li>
    <li>
      <a href="../articles/Typical_Processing_Workflow.html">Typical Processing Workflow</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/pmartR/pmartR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Reduce Dimension of Data for Exploratory Data Analysis</h1>
    <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/HEAD/R/dim_reduction.R" class="external-link"><code>R/dim_reduction.R</code></a></small>
    <div class="hidden name"><code>dim_reduction.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For data types other than seqData, this function calculates principal
components using projection pursuit estimation, which implements an
expectation-maximization (EM) estimation algorithm when data is missing. For
seqData counts, a generalized version of principal components analysis for
non-normally distributed data is calculated under the assumption of a
negative binomial distribution with global dispersion.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dim_reduction</span><span class="op">(</span><span class="va">omicsData</span>, k <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>omicsData</dt>
<dd><p>an object of the class 'pepdata', 'prodata', 'metabData',
'lipidData', 'nmrData', or 'seqData', created by <code><a href="as.pepData.html">as.pepData</a></code>,
<code><a href="as.proData.html">as.proData</a></code>, <code><a href="as.metabData.html">as.metabData</a></code>,
<code><a href="as.lipidData.html">as.lipidData</a></code>, <code><a href="as.nmrData.html">as.nmrData</a></code>, or
<code><a href="as.seqData.html">as.seqData</a></code>, respectively.</p></dd>


<dt>k</dt>
<dd><p>integer number of principal components to return. Defaults to 2.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a data.frame with first <code>k</code> principal component scores, sample
  identifiers, and group membership for each sample (if group designation was
  previously run on the data). The object is of class dimRes (dimension
  reduction Result).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Any biomolecules seen in only one sample or with a variance less
  than 1E-6 across all samples are not included in the PCA calculations. This
  function leverages code from <code><a href="https://rdrr.io/pkg/pcaMethods/man/pca.html" class="external-link">pca</a></code> and
  <code><a href="https://rdrr.io/pkg/glmpca/man/glmpca.html" class="external-link">glmpca</a></code> .</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Redestig H, Stacklies W, Scholz M, Selbig J, &amp; Walther
  D (2007). <em>pcaMethods - a bioconductor package providing PCA methods
  for incomplete data</em>. Bioinformatics. 23(9): 1164-7.</p>
<p>Townes FW, Hicks SC, Aryee MJ, Irizarry RA (2019). <em>Feature selection
  and dimension reduction for single-cell RNA-seq based on a multinomial
  model.</em> Genome Biol. 20, 1–16.</p>
<p>Huang H, Wang Y, Rudin C, Browne EP (2022). <em>Towards a comprehensive
  evaluation of dimension reduction methods for transcriptomic data
  visualization.</em> Communications Biology 5, 719.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pmartR/pmartRdata" class="external-link">pmartRdata</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">mylipid</span> <span class="op">&lt;-</span> <span class="fu"><a href="edata_transform.html">edata_transform</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">lipid_neg_object</span>, data_scale <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="va">mylipid</span> <span class="op">&lt;-</span> <span class="fu"><a href="group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mylipid</span>, main_effects <span class="op">=</span> <span class="st">"Virus"</span><span class="op">)</span></span>
<span><span class="va">pca_lipids</span> <span class="op">&lt;-</span> <span class="fu">dim_reduction</span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mylipid</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># \donttest{</span></span>
<span><span class="va">myseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">rnaseq_object</span>, main_effects <span class="op">=</span> <span class="st">"Virus"</span><span class="op">)</span></span>
<span><span class="va">pca_seq</span> <span class="op">&lt;-</span> <span class="fu">dim_reduction</span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">myseq</span><span class="op">)</span></span>
<span><span class="co"># }</span></span></code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Lisa Bramer, Kelly Stratton, Daniel Claborne.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

