<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Normalization • pmartR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data Normalization">
<meta property="og:description" content="pmartR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.4.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Convenience_Functions.html">Convenience Functions</a>
    </li>
    <li>
      <a href="../articles/Data_Normalization.html">Data Normalization</a>
    </li>
    <li>
      <a href="../articles/Data_Object_Creation.html">Data Object Creation</a>
    </li>
    <li>
      <a href="../articles/Filter_Functionality.html">Filter Functions</a>
    </li>
    <li>
      <a href="../articles/Graphing_with_pmartR.html">Graphing with pmartR</a>
    </li>
    <li>
      <a href="../articles/RNAseq_Processing_Workflow.html">RNAseq Data Processing</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Trelliscope_Vignette.html">Using Trelliscope Functions with pmartR</a>
    </li>
    <li>
      <a href="../articles/Typical_Processing_Workflow.html">Typical Processing Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pmartR/pmartR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Normalization</h1>
                        <h4 data-toc-skip class="author">Kelly Stratton,
Lisa Bramer</h4>
            
            <h4 data-toc-skip class="date">2023-12-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/HEAD/vignettes/Data_Normalization.Rmd" class="external-link"><code>vignettes/Data_Normalization.Rmd</code></a></small>
      <div class="hidden name"><code>Data_Normalization.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>There are two general types of normalization approaches included in
<code>pmartR</code>:</p>
<ol style="list-style-type: decimal">
<li><p>Instrument normalization - applicable to labeled proteomics data
and (sometimes) NMR data</p></li>
<li><p>Statistical normalization - applicable to all of the data types
supported in <code>pmartR</code>, with RNAseq data being a special
case</p></li>
</ol>
<p>When instrument normalization is required, this is done after quality
control (filtering) and prior to statistical normalization.</p>
</div>
<div class="section level2">
<h2 id="instrument-normalization">Instrument Normalization<a class="anchor" aria-label="anchor" href="#instrument-normalization"></a>
</h2>
<div class="section level3">
<h3 id="labeled-proteomics">Labeled Proteomics<a class="anchor" aria-label="anchor" href="#labeled-proteomics"></a>
</h3>
<p>Labeled proteomics data, such as that generated via TMT (tandem mass
tag) or iTRAQ (isobaric tags for relative quantification), involves
running subsets of samples from an experiment on separate “plexes” or
“plates” or “experiments”. Each plex includes a “reference pool” sample,
consisting of a pool of all of the experimental samples. (Note that the
assignment of samples to a plex is a key part of the randomization and
experimental design processes, due to the way data goes missing when
these labeled methods are utilized.) Samples must be normalized to their
corresponding reference pool in order to enable comparisons between
samples.</p>
<div class="section level4">
<h4 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h4>
<p>When working with labeled proteomics data, an
<code>isobaricpepData</code> object should be used. To perform the
reference pool normalization, use the <code><a href="../reference/normalize_isobaric.html">normalize_isobaric()</a></code>
function on data that has already been log transformed.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">isobaric_object</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">isobaric_object</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "isobaricpepData" "pepData"</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">isobaric_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span><span class="va">isobaric_object</span>,</span>
<span>  data_scale <span class="op">=</span> <span class="st">"log2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>There are two ways to specify the information needed for identifying
reference samples which should be used for normalization:</p>
<ol style="list-style-type: decimal">
<li>Specify the <code>refpool_cname</code> and
<code>refpool_notation</code>. This should be used when the reference
sample is not in a consistent channel across experiments/plates. Here,
<code>refpool_cname</code> gives the name of the column in
<code>f_data</code> which indicates whether a sample is a reference or
not, and <code>refpool_notation</code> is a character string giving the
value used to denote a reference sample in that column. } In both cases
you must specify <code>exp_cname</code> which gives the column name for
the column in <code>f_data</code> containing information about which
plex/plate/experiment a sample was run on.</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># do not apply the normalization until we have looked at the distribution of peptides within the reference pool samples</span></span>
<span><span class="va">iso_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_isobaric.html">normalize_isobaric</a></span><span class="op">(</span><span class="va">isobaric_object</span>,</span>
<span>  exp_cname <span class="op">=</span> <span class="st">"Plex"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  refpool_cname <span class="op">=</span> <span class="st">"Virus"</span>,</span>
<span>  refpool_notation <span class="op">=</span> <span class="st">"Pool"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># look at the distribution of peptides within the reference pool samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">iso_norm</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/specify_refpool-1.png" width="700"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># now apply the normalization</span></span>
<span><span class="va">isobaric_object_normalized</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_isobaric.html">normalize_isobaric</a></span><span class="op">(</span><span class="va">isobaric_object</span>,</span>
<span>  exp_cname <span class="op">=</span> <span class="st">"Plex"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  refpool_cname <span class="op">=</span> <span class="st">"Virus"</span>,</span>
<span>  refpool_notation <span class="op">=</span> <span class="st">"Pool"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># look at boxplots of the data after normalization to the reference pool samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">isobaric_object_normalized</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/specify_refpool-2.png" width="700"></p>
<ol start="2" style="list-style-type: decimal">
<li>Specify the <code>channel_cname</code> and
<code>refpool_channel</code>. This should be used when the reference
sample for each plex/plate/experiment was always located in the same
channel. Here <code>channel_cname</code> gives the column name for the
column in <code>f_data</code> which gives information about which
channel each sample was run on, and <code>refpool_channel</code> is a
character string specifying the value in <code>channel_colname</code>
that corresponds to the reference sample channel.</li>
</ol>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># not run because this data does not actually contain channel_cname or refpool_channel information</span></span>
<span><span class="va">iso_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_isobaric.html">normalize_isobaric</a></span><span class="op">(</span><span class="va">isobaric_object</span>,</span>
<span>  exp_cname <span class="op">=</span> <span class="st">"Plex"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  channel_cname <span class="op">=</span> <span class="st">"SampleChannel"</span>, <span class="co"># this column in f_data would specify a number corresponding to the channel for each sample</span></span>
<span>  refpool_channel <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span> <span class="co"># this value in the SampleChannel column would always correspond to a reference pool sample</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="nmr">NMR<a class="anchor" aria-label="anchor" href="#nmr"></a>
</h3>
<p>In order to ensure that samples are comparable to one another,
normalizing NMR data to either a sample-specific property (such as
sample mass) or to a spiked-in reference standard.</p>
<div class="section level4">
<h4 id="usage-1">Usage<a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h4>
<p>There are two ways to specify the information needed for performing
instrument normalization on an nmrData object:</p>
<ol style="list-style-type: decimal">
<li><p>Specify <code>sample_property_cname</code>. This should be used
when normalizing to a sample property, such as sample concentration or
mass, is desired. Here, <code>sample_property_cname</code> gives the
name of the column in <code>f_data</code> which contains the property to
use for normalization. If any samples have a missing value for this
column, and error is returned.</p></li>
<li><p>Specify <code>metabolite_name</code>. This should be used when
normalization to a spiked in standard is desired. Here
<code>metabolite_name</code> gives the name of the metabolite in e_data
(and e_meta, if present) corresponding to the spiked in standard. If any
samples have a missing value for this metabolite, an error is
returned.</p></li>
</ol>
<p>The option to backtransform the data after either of these NMR
normalizations is offered, if the user would like to ensure values are
on a scale similar to their raw values before normalization. The
following values are calculated and/or applied for backtransformation
purposes:</p>
<ul>
<li><p>If normalization using a metabolite in <code>e_data</code> is
specified, the location parameter is the median of the values for
<code>metabolite_name</code></p></li>
<li><p>If normalization using a sample property in <code>f_data</code>
is specified, the location parameter is the median of the values in
<code>sample_property</code></p></li>
</ul>
<p>We’ll use the example NMR dataset from the <code>pmartRdata</code>
package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">nmr_identified_object</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    </span></span>
<span><span class="co">## Class                       nmrData</span></span>
<span><span class="co">## Unique SampleIDs (f_data)        41</span></span>
<span><span class="co">## Unique Metabolites (e_data)      38</span></span>
<span><span class="co">## Unique Metabolites (e_meta)      38</span></span>
<span><span class="co">## Missing Observations              0</span></span>
<span><span class="co">## Proportion Missing                0</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># log2 transform the values</span></span>
<span><span class="va">nmr_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">nmr_identified_object</span>, data_scale <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Normalize using property of each sample</strong></p>
<p>We can first create a normRes object and explore the values of the
sample property to be used for instrument normalization. The figure
below shows the Concentration values for each sample.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># don't apply the normalization yet</span></span>
<span><span class="va">normRes_property</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/normalize_nmr.html">normalize_nmr</a></span><span class="op">(</span><span class="va">nmr_object</span>,</span>
<span>    apply_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    sample_property_cname <span class="op">=</span> <span class="st">"Concentration"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">normRes_property</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<p>If we are satisified with using these Concentration values for
normalization, we can apply the instrument normalization and plot the
resulting omicsData object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># now apply the normalization</span></span>
<span><span class="va">nmr_norm_property</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/normalize_nmr.html">normalize_nmr</a></span><span class="op">(</span><span class="va">nmr_object</span>,</span>
<span>    apply_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    sample_property_cname <span class="op">=</span> <span class="st">"Concentration"</span>,</span>
<span>    backtransform <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">nmr_norm_property</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p><strong>Normalize using reference metabolite</strong></p>
<p>We can first create a normRes object and explore the values of the
reference metabolite to be used for instrument normalization. The figure
below shows the log2 values of the reference metabolite for each
sample.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># don't apply the normalization yet</span></span>
<span><span class="va">normRes_reference</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/normalize_nmr.html">normalize_nmr</a></span><span class="op">(</span><span class="va">nmr_object</span>,</span>
<span>    apply_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    metabolite_name <span class="op">=</span> <span class="st">"unkm1.53"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">normRes_reference</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>If we are satisified with using these values for normalization, we
can apply the instrument normalization and plot the resulting omicsData
object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># now apply the normalization</span></span>
<span><span class="va">nmr_norm_reference</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_nmr.html">normalize_nmr</a></span><span class="op">(</span><span class="va">nmr_object</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  metabolite_name <span class="op">=</span> <span class="st">"unkm1.53"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## backtransform is set to FALSE. Examine the distribution of your data to ensure this is reasonable.</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">nmr_norm_reference</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="statistical-normalization">Statistical Normalization<a class="anchor" aria-label="anchor" href="#statistical-normalization"></a>
</h2>
<div class="section level3">
<h3 id="global-normalization">Global Normalization<a class="anchor" aria-label="anchor" href="#global-normalization"></a>
</h3>
<p>There are two specifications needed in order to compute normalization
factors: the normalization function (how the normalization factors are
computed) and the subset function (the subset of biomolecules with which
the normalization factors are computed) <span class="citation">(Åstrand
2003)</span>,<span class="citation">(Gautier et al. 2004)</span>. Global
normalizations in <code>pmartR</code> normalize within each sample, so
each sample has its own normalization factor(s), a location and
(sometimes) a scale parameter. There are several options available for
normalization and subset functions.</p>
<p><strong>Normalization Functions</strong></p>
<p>For all of the normalization functions, <code>na.rm = TRUE</code> is
utilized when computing the parameters.</p>
<ul>
<li><p>Median: Location parameter for each sample is equal to the median
of the observed biomolecules in the subset specified. There is no scale
parameter.</p></li>
<li><p>Mean: Location parameter for each sample is equal to the mean of
the observed biomolecules in the subset specified. There is no scale
parameter.</p></li>
<li><p>Median Absolute Deviation (MAD): Location parameter for each
sample is equal to the median of the observed biomolecules in the subset
specified. Scale parameter is equal to the MAD of the observed
biomolecules in the subset specified.</p></li>
<li><p>Z-score: Location parameter for each sample is equal to the mean
of the observed biomolecules in the subset specified. Scale parameter is
equal to the standard deviation of the observed biomolecules in the
subset specified.</p></li>
</ul>
<p><strong>Subset Functions</strong></p>
<ul>
<li><p>All: All biomolecules are utilized</p></li>
<li><p>L-Order Statistics (LOS): Identifies the subset of the
biomolecules associated with the top L order statistics, where L is a
proportion between 0 and 1. Specifically, the biomolecules with the top
L proportion of highest absolute abundance are retained for each sample,
and the union of these biomolecules is taken as the subset identified
(Wang et al., 2006)</p></li>
<li><p>Percentage of Peptides Present (PPP): Originally developed for
use with proteomics data, this subset can also be applied to other omics
data types. PPP identifies the subset of biomolecules that are
present/non-missing for a minimum proportion of samples (Karpievitch et
al., 2009; Kultima et al., 2009).</p></li>
<li><p>Rank Invariant Peptides (RIP): Identifies biomolecules with
complete data that have a p-value greater than a defined threshold alpha
(common values include 0.1 or 0.25) when subjected to a Kruskal-Wallis
test based (non-parametric one-way ANOVA) on group membership
(Webb-Robertson et al., 2011).</p></li>
<li><p>PPP-RIP: Equivalent to RIP, however, rather than requiring
biomolecules with complete data, biomolecules with at least a specified
proportion of non-missing values are subject to the Kruskal-Wallis
test.</p></li>
<li><p>Complete Peptides: Biomolecules with no missing data across all
samples, equivalent to PPP with proportion = 1.</p></li>
</ul>
<div class="section level4">
<h4 id="usage-2">Usage<a class="anchor" aria-label="anchor" href="#usage-2"></a>
</h4>
<p>Global normalization methods are applied with the
<code><a href="../reference/normalize_global.html">normalize_global()</a></code> function. There is an argument to the
function, <code>apply_norm</code>, that dictates whether the
normalization method is actually applied to the data
(<code>apply_norm = TRUE)</code> or if instead the normalization factors
are computed so that the user can assess the appropriateness of the
normalization approach prior to applying it to a dataset
(<code>apply_norm = FALSE)</code>. Similar to NMR normalization, the
option to backtransform the data after global normalization is available
in order to put the normalized values back on roughly the same scale as
the raw (un-normalized) data (<code>backtransform = TRUE</code>).</p>
<p>Global median centering is our go-to approach, especially for LC-MS
lipidomics, GC-MS metabolomics, and NMR. Below we provide a variety of
examples of using the <code><a href="../reference/normalize_global.html">normalize_global()</a></code> function with
different parameter values.</p>
<p>When, in preparation for normalizing the data, we use the
<code><a href="../reference/normalize_global.html">normalize_global()</a></code> function but do not apply the
normalization, we get back an object of class normRes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># global median centering - don't apply the norm, show info in the normRes object</span></span>
<span><span class="va">mypep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">pep_object</span>, data_scale <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mynorm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "normRes"</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># plot(mynorm)</span></span></code></pre></div>
<p>When we apply the normalization to the data, we get back an omicsData
object of the same class that we started with. The
<code>backtransform</code> argument can be set to <code>TRUE</code> to
return the normalized data such that the normalized values are similar
in magnitude to the pre-normalized values. With global median centering,
this is done by adding back the median of the entire dataset to each
individual value; this results in boxplots that are no longer centered
at zero, but rather at the median of the entire dataset.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># global median centering - apply the norm</span></span>
<span><span class="va">mypep_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  backtransform <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mypep_norm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "pepData"</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mypep_norm</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>When using a subset function other than “all”, additional parameters
are needed; they have default values or can be set by the user. Here we
use rank invariant biomolecules (RIP) with a parameter of 0.2. For
details, see <code><a href="../reference/normalize_global.html">?normalize_global</a></code>. Note that some of the
subset functions, such as “rip”, require <code>group_designation</code>
to have been run on the omicsData object.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># run group_designation</span></span>
<span><span class="va">mypep</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep</span>, main_effects <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># RIP mean centering - don't apply the norm (so we can look at the number of molecules in the subset)</span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"rip"</span>,</span>
<span>  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>rip <span class="op">=</span> <span class="op">(</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># we can see how many biomolecules are included in this subset</span></span>
<span><span class="va">mynorm</span><span class="op">$</span><span class="va">n_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2746</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mynorm</span><span class="op">$</span><span class="va">prop_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.1287449</span></span></code></pre>
<p>If the subset of biomolecules used does not contain a sufficient
number or proportion, we have less confidence in the normalization and
may want to choose a different subset function. As a rule of thumb, we
like to see at least 10% of the biomolecules from the
normalization-ready dataset utilized in the subset for calculating the
normalization parameter(s). If the normalization-ready dataset has fewer
than ~200 biomolecules, as is common with LC-MS lipidomics and GC-MS
metabolomics, 10% of the biomolecules may still not be enough to have
confidence in the normalization, as this is only ~20 biomolecules.</p>
<p>We can use the <code>min_prop</code> argument to the
<code>normalize_global</code> function to enforce a minimum proportion
of biomolecules required to apply a normalization. In the following code
using the LOS (0.1) subset, we would get an error if we included
<code>min_prop = 0.2</code> because the proportion of biomolecules in
the specified subset is below that threshold.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># LOS MAD</span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"los"</span>,</span>
<span>  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>los <span class="op">=</span> <span class="op">(</span><span class="fl">0.1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"mad"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mynorm</span><span class="op">$</span><span class="va">prop_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.1821933</span></span></code></pre>
<p>Here we demonstrate how to use the PPP subset and z-score
normalization function.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># PPP z-score</span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"ppp"</span>,</span>
<span>  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>ppp <span class="op">=</span> <span class="op">(</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"zscore"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  min_prop <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mynorm</span><span class="op">$</span><span class="va">prop_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5600825</span></span></code></pre>
<p>Here we demonstrate how to use the PPP-RIP subset and specify
parameter values, along with the median normalization function.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># PPP-RIP median</span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"ppp_rip"</span>,</span>
<span>  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>ppp <span class="op">=</span> <span class="op">(</span><span class="fl">0.5</span><span class="op">)</span>, rip <span class="op">=</span> <span class="op">(</span><span class="fl">0.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mynorm</span><span class="op">$</span><span class="va">prop_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.4433869</span></span></code></pre>
<p>Finally, we demonstrate how to use the set of complete biomolecules
with the mean function.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Complete mean</span></span>
<span><span class="va">mynorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mynorm</span><span class="op">$</span><span class="va">prop_features_calc</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.1581415</span></span></code></pre>
<p>For proteomics data, or large lipidomics or metabolomics datasets
(&gt;200ish biomolecules??), the Statistical Procedure for the Analyses
of peptide abundance Normalization Strategies (SPANS) <span class="citation">(Webb-Robertson et al. 2011)</span> algorithm can help
determine an appropriate normalization approach. This algorithm ranks
different combinations of subset and normalization functions based on a
score that captures how much bias a particular normalization procedure
introduces into the data. Higher scores imply less bias.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># returns a data frame arranged by descending SPANS score</span></span>
<span><span class="co"># not run due to long runtime; SPANS plot generated and saved to include in vignette</span></span>
<span><span class="va">spans_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/spans_procedure.html">spans_procedure</a></span><span class="op">(</span><span class="va">mypep</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">spans_result</span><span class="op">)</span></span></code></pre></div>
<p><img src="SPANS_pep.png" width="600px"></p>
<p>To apply a normalization method after using SPANS, simply use the
<code>normalize_global</code> function.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_global.html">normalize_global</a></span><span class="op">(</span></span>
<span>  omicsData <span class="op">=</span> <span class="va">mypep</span>,</span>
<span>  subset_fn <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  norm_fn <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  apply_norm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  backtransform <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="loess-normalization">LOESS Normalization<a class="anchor" aria-label="anchor" href="#loess-normalization"></a>
</h3>
<p>As an alternative to a global normalization, <code>pmartR</code>
includes a wrapper for the <code>normalizeCyclicLoess</code> <span class="citation">(Bolstad et al. 2003)</span> function from the limma
package <span class="citation">(Smyth 2005)</span>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mypep</span>, order_by <span class="op">=</span> <span class="st">"Phenotype"</span>, color_by <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_loess</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_loess.html">normalize_loess</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mypep_loess</span>, order_by <span class="op">=</span> <span class="st">"Phenotype"</span>, color_by <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="quantile-normalization">Quantile Normalization<a class="anchor" aria-label="anchor" href="#quantile-normalization"></a>
</h3>
<p>Quantile normalization is an algorithm for normalizing a set of data
vectors by giving them the same distribution <span class="citation">(Bolstad et al. 2003)</span>. It is applied to data on
the abundance scale (e.g. not a log scale). It is often used for
microarray data. Note that quantile normalization requires complete
data. In the example below we use the <code>molecule_filter</code> to
restrict our data to complete samples.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mymetab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">metab_object</span>, data_scale <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># restrict to complete samples using the molecule filter</span></span>
<span><span class="va">myfilter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mymetab</span><span class="op">)</span></span>
<span><span class="va">nsamps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_data_info.html">get_data_info</a></span><span class="op">(</span><span class="va">mymetab</span><span class="op">)</span><span class="op">$</span><span class="va">num_samps</span></span>
<span><span class="va">mymetab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myfilter</span>, omicsData <span class="op">=</span> <span class="va">mymetab</span>, min_num <span class="op">=</span> <span class="va">nsamps</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mymetab</span>, order_by <span class="op">=</span> <span class="st">"Phenotype"</span>, color_by <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mymetab_quantile</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_quantile.html">normalize_quantile</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mymetab</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mymetab_quantile</span>, order_by <span class="op">=</span> <span class="st">"Phenotype"</span>, color_by <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Data_Normalization_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
<p>It is up to the user to determine whether the normalization
ultimately applied to the data is appropriate. In the case of the
quantile normalization performed above, based on the resulting boxplots
we would not recommend proceeding with this dataset for further
analyses; a different normalization approach should be tried.</p>
</div>
<div class="section level3">
<h3 id="rnaseq-data-normalization">RNAseq Data Normalization<a class="anchor" aria-label="anchor" href="#rnaseq-data-normalization"></a>
</h3>
<p>The statistical normalization approaches for RNAseq data are combined
with the statistical analysis functions. The following are supported
within <code>pmartR</code> via wrapper functions for the original
functionality of the corresponding packages.</p>
<ul>
<li><p>DESeq2 <span class="citation">(Love, Anders, and Huber
2014)</span></p></li>
<li><p>edgeR <span class="citation">(Robinson, McCarthy, and Smyth
2010)</span></p></li>
<li><p>limma voom <span class="citation">(Law et al.
2014)</span></p></li>
</ul>
<p>See <code><a href="../reference/diffexp_seq.html">?diffexp_seq</a></code> and the “RNAseq Processing Workflow”
for more information.</p>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-aastrand2003contrast" class="csl-entry">
Åstrand, Magnus. 2003. <span>“Contrast Normalization of Oligonucleotide
Arrays.”</span> <em>Journal of Computational Biology</em> 10 (1):
95–102.
</div>
<div id="ref-bolstad2003comparison" class="csl-entry">
Bolstad, Benjamin M, Rafael A Irizarry, Magnus Åstrand, and Terence P.
Speed. 2003. <span>“A Comparison of Normalization Methods for High
Density Oligonucleotide Array Data Based on Variance and Bias.”</span>
<em>Bioinformatics</em> 19 (2): 185–93.
</div>
<div id="ref-gautier2004affy" class="csl-entry">
Gautier, Laurent, Leslie Cope, Benjamin M Bolstad, and Rafael A
Irizarry. 2004. <span>“Affy—Analysis of Affymetrix GeneChip Data at the
Probe Level.”</span> <em>Bioinformatics</em> 20 (3): 307–15.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014.
<span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for
RNA-Seq Read Counts.”</span> <em>Genome Biology</em> 15 (2): 1–17.
</div>
<div id="ref-love2014differential" class="csl-entry">
Love, Michael, Simon Anders, and Wolfgang Huber. 2014.
<span>“Differential Analysis of Count Data–the DESeq2 Package.”</span>
<em>Genome Biol</em> 15 (550): 10–1186.
</div>
<div id="ref-robinson2010edger" class="csl-entry">
Robinson, Mark D, Davis J McCarthy, and Gordon K Smyth. 2010.
<span>“edgeR: A Bioconductor Package for Differential Expression
Analysis of Digital Gene Expression Data.”</span>
<em>Bioinformatics</em> 26 (1): 139–40.
</div>
<div id="ref-smyth2005limma" class="csl-entry">
Smyth, Gordon K. 2005. <span>“Limma: Linear Models for Microarray
Data.”</span> In <em>Bioinformatics and Computational Biology Solutions
Using r and Bioconductor</em>, 397–420. Springer.
</div>
<div id="ref-webb2011statistical" class="csl-entry">
Webb-Robertson, Bobbie-Jo M, Melissa M Matzke, Jon M Jacobs, Joel G
Pounds, and Katrina M Waters. 2011. <span>“A Statistical Selection
Strategy for Normalization Procedures in LC-MS Proteomics Experiments
Through Dataset-Dependent Ranking of Normalization Scaling
Factors.”</span> <em>Proteomics</em> 11 (24): 4736–41.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Lisa Bramer, Kelly Stratton, Daniel Claborne.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
