<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Filter Functions • pmartR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Filter Functions">
<meta property="og:description" content="pmartR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pmartR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Convenience_Functions.html">Convenience Functions</a>
    </li>
    <li>
      <a href="../articles/Data_Normalization.html">Data Normalization</a>
    </li>
    <li>
      <a href="../articles/Data_Object_Creation.html">Data Object Creation</a>
    </li>
    <li>
      <a href="../articles/Filter_Functionality.html">Filter Functions</a>
    </li>
    <li>
      <a href="../articles/Graphing_with_pmartR.html">Graphing with pmartR</a>
    </li>
    <li>
      <a href="../articles/RNAseq_Processing_Workflow.html">RNAseq Data Processing</a>
    </li>
    <li>
      <a href="../articles/Statistical_Analysis_with_pmartR.html">Statistical Analysis with the `pmartR` Package</a>
    </li>
    <li>
      <a href="../articles/Trelliscope_Vignette.html">Using Trelliscope Functions with pmartR</a>
    </li>
    <li>
      <a href="../articles/Typical_Processing_Workflow.html">Typical Processing Workflow</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pmartR/pmartR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Filter Functions</h1>
                        <h4 data-toc-skip class="author">Kelly Stratton,
Lisa Bramer</h4>
            
            <h4 data-toc-skip class="date">2023-03-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pmartR/pmartR/blob/HEAD/vignettes/Filter_Functionality.Rmd" class="external-link"><code>vignettes/Filter_Functionality.Rmd</code></a></small>
      <div class="hidden name"><code>Filter_Functionality.Rmd</code></div>

    </div>

    
    
<p><img src="pmartR_logo_final.jpg" width="100px"></p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This vignette describes the filter functionality of the
<code>pmartR</code> package. Filters fall into three categories and
their use is a standard part of quality control processing.</p>
<ol style="list-style-type: decimal">
<li><p>Biomolecule filters</p></li>
<li><p>Sample filters</p></li>
<li><p>Filters specific to seqData objects</p></li>
</ol>
<p>In a typical <code>pmartR</code> processing pipeline, these filters
are applied in a certain order during the quality control process.</p>
<p>To utilize a filter, the first step is to create the filter object,
an S3 object whose class is directly tied to the filter. Once a filter
object is created, the <code>plot</code> and <code>summary</code>
functions can be used to see the effects of applying the filter with
certain parameters, and then the filter can be applied to the omicsData
object using the <code>applyFilt</code> function.</p>
<p>If data are paired, the pairing structure can be defined in the
<code>group_designation</code> function, and the pairs will be accounted
for in relevant filters. For biomolecule filters, this means filtering
based on paired differences rather than on individual values. For sample
filters, removal of a sample will be accompanied by the removal of all
members of the corresponding pair.</p>
</div>
<div class="section level2">
<h2 id="biomolecule-filters">Biomolecule Filters<a class="anchor" aria-label="anchor" href="#biomolecule-filters"></a>
</h2>
<p>The following filters remove entire biomolecules from omicsData
objects.</p>
<div class="section level3">
<h3 id="molecule-filter">Molecule Filter<a class="anchor" aria-label="anchor" href="#molecule-filter"></a>
</h3>
<p>The motivation for using the molecule filter is to remove any
biomolecules not confidently observed. Typical use of the molecule
filter removes any biomolecules that are not observed in at least two
samples across the entire dataset, within each group, or within each
batch. Depending on downstream analysis goals, a different threshold may
be more appropriate.</p>
<p>Below we demonstrate creation and application of the molecule filter
on a pepData object. With a minimum number of 2 observations per
molecule required across all samples, a summary of the molecule filter
object tells us how many biomolecules would be filtered and not filtered
out of the dataset.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep</span> <span class="op">&lt;-</span> <span class="va">pep_object</span></span>
<span></span>
<span><span class="co"># create a molecule filter object</span></span>
<span><span class="va">mymolfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mymolfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "moleculeFilt" "data.frame"</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1              930</span></span>
<span><span class="co">##                 2             1928</span></span>
<span><span class="co">##                 3             2895</span></span>
<span><span class="co">##                 4             3826</span></span>
<span><span class="co">##                 5             4705</span></span>
<span><span class="co">##                 6             5627</span></span>
<span><span class="co">##                 7             6443</span></span>
<span><span class="co">##                 8             7213</span></span>
<span><span class="co">##                 9             7937</span></span>
<span><span class="co">##                10             8647</span></span>
<span><span class="co">##                11             9383</span></span>
<span><span class="co">##                12            10032</span></span>
<span><span class="co">##                13            10678</span></span>
<span><span class="co">##                14            11292</span></span>
<span><span class="co">##                15            11937</span></span>
<span><span class="co">##                16            12592</span></span>
<span><span class="co">##                17            13240</span></span>
<span><span class="co">##                18            13917</span></span>
<span><span class="co">##                19            14571</span></span>
<span><span class="co">##                20            15248</span></span>
<span><span class="co">##                21            15941</span></span>
<span><span class="co">##                22            16800</span></span>
<span><span class="co">##                23            17956</span></span>
<span><span class="co">##                24            21329</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># summary of molecule filter with a specific threshold</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## Minimum Number:2</span></span>
<span><span class="co">## Filtered:930</span></span>
<span><span class="co">## Not Filtered:20399</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1              930</span></span>
<span><span class="co">##                 2             1928</span></span>
<span><span class="co">##                 3             2895</span></span>
<span><span class="co">##                 4             3826</span></span>
<span><span class="co">##                 5             4705</span></span>
<span><span class="co">##                 6             5627</span></span>
<span><span class="co">##                 7             6443</span></span>
<span><span class="co">##                 8             7213</span></span>
<span><span class="co">##                 9             7937</span></span>
<span><span class="co">##                10             8647</span></span>
<span><span class="co">##                11             9383</span></span>
<span><span class="co">##                12            10032</span></span>
<span><span class="co">##                13            10678</span></span>
<span><span class="co">##                14            11292</span></span>
<span><span class="co">##                15            11937</span></span>
<span><span class="co">##                16            12592</span></span>
<span><span class="co">##                17            13240</span></span>
<span><span class="co">##                18            13917</span></span>
<span><span class="co">##                19            14571</span></span>
<span><span class="co">##                20            15248</span></span>
<span><span class="co">##                21            15941</span></span>
<span><span class="co">##                22            16800</span></span>
<span><span class="co">##                23            17956</span></span>
<span><span class="co">##                24            21329</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_molfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span><span class="va">mymolfilt</span>, omicsData <span class="op">=</span> <span class="va">mypep</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now our data object contains the reduced number of biomolecules</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mypep_molfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      </span></span>
<span><span class="co">## Class                         pepData</span></span>
<span><span class="co">## Unique SampleIDs (f_data)          24</span></span>
<span><span class="co">## Unique Peptides (e_data)        20399</span></span>
<span><span class="co">## Unique RazorProteins (e_meta)    2916</span></span>
<span><span class="co">## Missing Observations           202348</span></span>
<span><span class="co">## Proportion Missing              0.413</span></span></code></pre>
<p>To implement a molecule filter with the minimum required number of
observations is within each group, we must first define groups.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define groups based on a "main effect"</span></span>
<span><span class="va">mypep_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep</span>, main_effects <span class="op">=</span> <span class="st">"Phenotype"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a molecule filter object</span></span>
<span><span class="va">mymolfilt_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups</span>, use_groups <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mymolfilt_groups</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "moleculeFilt" "data.frame"</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt_groups</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Minimum Number is Minimum Number per Group </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1             6885</span></span>
<span><span class="co">##                 2             9152</span></span>
<span><span class="co">##                 3            11013</span></span>
<span><span class="co">##                 4            12797</span></span>
<span><span class="co">##                 5            14448</span></span>
<span><span class="co">##                 6            16036</span></span>
<span><span class="co">##                 7            17956</span></span>
<span><span class="co">##                 8            21329</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># summary of molecule filter with a specific threshold</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt_groups</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Minimum Number is Minimum Number per Group </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## Minimum Number:2</span></span>
<span><span class="co">## Filtered:6885</span></span>
<span><span class="co">## Not Filtered:14444</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1             6885</span></span>
<span><span class="co">##                 2             9152</span></span>
<span><span class="co">##                 3            11013</span></span>
<span><span class="co">##                 4            12797</span></span>
<span><span class="co">##                 5            14448</span></span>
<span><span class="co">##                 6            16036</span></span>
<span><span class="co">##                 7            17956</span></span>
<span><span class="co">##                 8            21329</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_molfilt_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span><span class="va">mymolfilt_groups</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now our data object contains the reduced number of biomolecules</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mypep_molfilt_groups</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      </span></span>
<span><span class="co">## Class                         pepData</span></span>
<span><span class="co">## Unique SampleIDs (f_data)          24</span></span>
<span><span class="co">## Unique Peptides (e_data)        14444</span></span>
<span><span class="co">## Unique RazorProteins (e_meta)    2389</span></span>
<span><span class="co">## Missing Observations            88899</span></span>
<span><span class="co">## Proportion Missing              0.256</span></span>
<span><span class="co">## Samples per group: Phenotype3       8</span></span>
<span><span class="co">## Samples per group: Phenotype1       8</span></span>
<span><span class="co">## Samples per group: Phenotype2       8</span></span></code></pre>
<p>If our samples were run on the instrument over the course of multiple
batches, we may wish to implement the molecule filter within each
batch.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># define groups based on a "main effect" and "batch id" -- here we pretend that "SecondPhenotype" is describing the batches of samples</span></span>
<span><span class="va">mypep_batches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/group_designation.html">group_designation</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep</span>, main_effects <span class="op">=</span> <span class="st">"Phenotype"</span>, batch_id <span class="op">=</span> <span class="st">"SecondPhenotype"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create a molecule filter object using the batch information -- we can use both groups and batch or just batch</span></span>
<span><span class="va">mymolfilt_batches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_batches</span>, use_groups <span class="op">=</span> <span class="cn">TRUE</span>, use_batch <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">mymolfilt_batches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/molecule_filter.html">molecule_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_batches</span>, use_batch <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mymolfilt_batches</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "moleculeFilt" "data.frame"</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt_batches</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Minimum Number is Minimum Number per Batch </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1             4198</span></span>
<span><span class="co">##                 2             6104</span></span>
<span><span class="co">##                 3             7742</span></span>
<span><span class="co">##                 4             9252</span></span>
<span><span class="co">##                 5            10522</span></span>
<span><span class="co">##                 6            11679</span></span>
<span><span class="co">##                 7            12860</span></span>
<span><span class="co">##                 8            14067</span></span>
<span><span class="co">##                 9            15171</span></span>
<span><span class="co">##                10            16376</span></span>
<span><span class="co">##                11            17956</span></span>
<span><span class="co">##                12            21329</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># summary of molecule filter with a specific threshold</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mymolfilt_batches</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Molecule Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Minimum Number is Minimum Number per Batch </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## Minimum Number:2</span></span>
<span><span class="co">## Filtered:4198</span></span>
<span><span class="co">## Not Filtered:17131</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##  num_observations frequency_counts</span></span>
<span><span class="co">##                 1             4198</span></span>
<span><span class="co">##                 2             6104</span></span>
<span><span class="co">##                 3             7742</span></span>
<span><span class="co">##                 4             9252</span></span>
<span><span class="co">##                 5            10522</span></span>
<span><span class="co">##                 6            11679</span></span>
<span><span class="co">##                 7            12860</span></span>
<span><span class="co">##                 8            14067</span></span>
<span><span class="co">##                 9            15171</span></span>
<span><span class="co">##                10            16376</span></span>
<span><span class="co">##                11            17956</span></span>
<span><span class="co">##                12            21329</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_molfilt_batches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span><span class="va">mymolfilt_batches</span>, omicsData <span class="op">=</span> <span class="va">mypep_batches</span>, min_num <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now our data object contains the reduced number of biomolecules</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mypep_molfilt_batches</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      </span></span>
<span><span class="co">## Class                         pepData</span></span>
<span><span class="co">## Unique SampleIDs (f_data)          24</span></span>
<span><span class="co">## Unique Peptides (e_data)        17131</span></span>
<span><span class="co">## Unique RazorProteins (e_meta)    2650</span></span>
<span><span class="co">## Missing Observations           135345</span></span>
<span><span class="co">## Proportion Missing              0.329</span></span>
<span><span class="co">## Samples per group: Phenotype3       8</span></span>
<span><span class="co">## Samples per group: Phenotype1       8</span></span>
<span><span class="co">## Samples per group: Phenotype2       8</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="coefficient-of-variation-cv-filter">Coefficient of Variation (CV) Filter<a class="anchor" aria-label="anchor" href="#coefficient-of-variation-cv-filter"></a>
</h3>
<p>The coefficient of variation (CV) is the ratio of the standard
deviation to the mean <span class="citation">(Ahmed 1995)</span>. We can
use the CV calculated for each biomolecule as a quality control filter
to remove highly variable biomolecules. Using the pooled CV, as
<code>pmartR</code> does, enables the experimental groups to be
accounted for. Note that the CV is calculated on the abundance scale (if
already log transformed, the <code><a href="../reference/cv_filter.html">cv_filter()</a></code> function accounts
for this).</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use the example peptide data with Phenotype as main effect</span></span>
<span><span class="va">mycvfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cv_filter.html">cv_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mycvfilt</span>, cv_threshold <span class="op">=</span> <span class="fl">97</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/cv%20filter-1.png" width="768"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we get the same graph if we log2 transform our data</span></span>
<span><span class="va">mypep_groups_log2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edata_transform.html">edata_transform</a></span><span class="op">(</span><span class="va">mypep_groups</span>, data_scale <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="va">mycvfilt_log2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cv_filter.html">cv_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mycvfilt_log2</span>, cv_threshold <span class="op">=</span> <span class="fl">97</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/cv%20filter-2.png" width="768"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mycvfilt_log2</span>, cv_threshold <span class="op">=</span> <span class="fl">97</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Coefficient of Variation (CV) Filter</span></span>
<span><span class="co">## ----------------------</span></span>
<span><span class="co">## CVs:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. </span></span>
<span><span class="co">##   0.03514  27.54125  35.54875  37.42133  45.55902 171.45156 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Total NAs: 1866</span></span>
<span><span class="co">## Total Non-NAs: 19463 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Number Filtered Biomolecules: 83</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># apply the filter</span></span>
<span><span class="va">mypep_cvfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">mycvfilt_log2</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, cv_threshold <span class="op">=</span> <span class="fl">97</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mypep_cvfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                      </span></span>
<span><span class="co">## Class                         pepData</span></span>
<span><span class="co">## Unique SampleIDs (f_data)          24</span></span>
<span><span class="co">## Unique Peptides (e_data)        21246</span></span>
<span><span class="co">## Unique RazorProteins (e_meta)    2979</span></span>
<span><span class="co">## Missing Observations           222584</span></span>
<span><span class="co">## Proportion Missing              0.437</span></span>
<span><span class="co">## Samples per group: Phenotype3       8</span></span>
<span><span class="co">## Samples per group: Phenotype1       8</span></span>
<span><span class="co">## Samples per group: Phenotype2       8</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="proteomics-filter">Proteomics Filter<a class="anchor" aria-label="anchor" href="#proteomics-filter"></a>
</h3>
<p>For proteomics data (pepData object where mapping of
peptides-to-proteins is included in e_meta), two additional filters are
available via the <code><a href="../reference/proteomics_filter.html">proteomics_filter()</a></code> function.</p>
<div class="section level4">
<h4 id="redundant-peptide-filter">Redundant Peptide Filter<a class="anchor" aria-label="anchor" href="#redundant-peptide-filter"></a>
</h4>
<p>In some instances, peptides can map to more than one protein;
peptides such as this are called “redundant” or “degenerate”. Redundant
peptides can be removed using the <code><a href="../reference/proteomics_filter.html">proteomics_filter()</a></code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myproteomicsfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/proteomics_filter.html">proteomics_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myproteomicsfilt</span>, degen_peps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Proteomics Filter </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##              Obs Proteins Per Peptide Obs Peptides Per Protein</span></span>
<span><span class="co">## Min.                                1                        1</span></span>
<span><span class="co">## 1st Qu.                             1                        2</span></span>
<span><span class="co">## Median                              1                        3</span></span>
<span><span class="co">## Mean                                1         7.15258215962441</span></span>
<span><span class="co">## 3rd Qu.                             1                        8</span></span>
<span><span class="co">## Max.                                1                      235</span></span>
<span><span class="co">##                                                               </span></span>
<span><span class="co">## Filtered                            0                        0</span></span>
<span><span class="co">## Not Filtered                    21329                     2982</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This results in an error message since there are no redundant peptides</span></span>
<span><span class="co"># mypep_redundant &lt;- applyFilt(filter_object = myredundantfilt, omicsData = mypep_groups_log2)</span></span></code></pre></div>
<p>Here we have no redundant peptides, so a summary of the filter object
indicates that no peptides or proteins are removed. Trying to apply the
filter in this scenario results in an error.</p>
</div>
<div class="section level4">
<h4 id="minimum-number-of-peptides-per-protein-filter">Minimum Number of Peptides per Protein Filter<a class="anchor" aria-label="anchor" href="#minimum-number-of-peptides-per-protein-filter"></a>
</h4>
<p>Sometimes it is of interest to remove any proteins that are not
“confidently” observed, where “confidently” is defined by the number of
peptides observed that map to a given protein. For instance, we could
use the following filter to remove any proteins (and their associated
peptides) that have fewer than 2 peptides mapping to them.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use the same filter object as above</span></span>
<span><span class="co"># setting min_num_peps = 2 removes any proteins with just a single observed peptide mapping to them; the associated peptides are also removed</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myproteomicsfilt</span>, min_num_peps <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Proteomics Filter </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##              Obs Proteins Per Peptide Obs Peptides Per Protein</span></span>
<span><span class="co">## Min.                                1                        1</span></span>
<span><span class="co">## 1st Qu.                             1                        2</span></span>
<span><span class="co">## Median                              1                        3</span></span>
<span><span class="co">## Mean                                1         7.15258215962441</span></span>
<span><span class="co">## 3rd Qu.                             1                        8</span></span>
<span><span class="co">## Max.                                1                      235</span></span>
<span><span class="co">##                                                               </span></span>
<span><span class="co">## Filtered                            0                      664</span></span>
<span><span class="co">## Not Filtered                    21329                     2318</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_proteomicsfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myproteomicsfilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, min_num_peps <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="imd-anova-filter">IMD-ANOVA Filter<a class="anchor" aria-label="anchor" href="#imd-anova-filter"></a>
</h3>
<p>If ANOVA and/or the IMD test will be used for downstream statistical
comparisons, any biomolecules for which the desired test cannot be run
can be removed using the following filter.</p>
<p>In the case where both ANOVA and IMD will be performed:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myimdanovafilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/imdanova_filter.html">imdanova_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myimdanovafilt</span>, min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span>, min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of IMD-ANOVA Filter</span></span>
<span><span class="co">##                           </span></span>
<span><span class="co">## Total Observations:  21329</span></span>
<span><span class="co">## Filtered:             3223</span></span>
<span><span class="co">## Not Filtered:        18106</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_imdanovafilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myimdanovafilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span>, min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## You have specified remove_singleton_groups = TRUE, but there are no singleton groups to remove. Proceeding with application of the IMD-ANOVA filter.</span></span></code></pre>
<p>If only ANOVA is of interest:</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># start with the same filter object as above</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myimdanovafilt</span>, min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of IMD-ANOVA Filter</span></span>
<span><span class="co">##                           </span></span>
<span><span class="co">## Total Observations:  21329</span></span>
<span><span class="co">## Filtered:             4129</span></span>
<span><span class="co">## Not Filtered:        17200</span></span></code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_anovafilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myimdanovafilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, min_nonmiss_anova <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## You have specified remove_singleton_groups = TRUE, but there are no singleton groups to remove. Proceeding with application of the IMD-ANOVA filter.</span></span></code></pre>
<p>If only the IMD test, also referred to as the g-test, is of
interest:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># start with the same filter object as above</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myimdanovafilt</span>, min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of IMD-ANOVA Filter</span></span>
<span><span class="co">##                           </span></span>
<span><span class="co">## Total Observations:  21329</span></span>
<span><span class="co">## Filtered:             3899</span></span>
<span><span class="co">## Not Filtered:        17430</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_imdfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myimdanovafilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, min_nonmiss_gtest <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## You have specified remove_singleton_groups = TRUE, but there are no singleton groups to remove. Proceeding with application of the IMD-ANOVA filter.</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="custom-filter">Custom Filter<a class="anchor" aria-label="anchor" href="#custom-filter"></a>
</h3>
<p>If certain biomolecules are known <em>a priori</em> to be unwanted in
the analysis, those can be removed using a custom filter. For instance
if contaminant proteins are identified, or a set of internal standards
needs to be removed for a specific portion of the analysis, this filter
can be of particular use. Multiple custom filters could be applied to a
single dataset, depending on the analysis needs. Biomolecules for
removal can be specified based on their presence in e_data or in
e_meta.</p>
<p>To remove biomolecules based on their presence in e_data, we
implement the filter as follows. Note that this also removes them from
e_meta, if applicable.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">remove_peps</span> <span class="op">&lt;-</span> <span class="va">mypep</span><span class="op">$</span><span class="va">e_data</span><span class="op">$</span><span class="va">Peptide</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="va">mycustomfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/custom_filter.html">custom_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, e_data_remove <span class="op">=</span> <span class="va">remove_peps</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mycustomfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Custom Filter</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                        Filtered Remaining Total</span></span>
<span><span class="co">## SampleIDs (f_data)            0        24    24</span></span>
<span><span class="co">## Peptides (e_data)            10     21319 21329</span></span>
<span><span class="co">## RazorProteins (e_meta)        2      2980  2982</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_custom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">mycustomfilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span><span class="op">)</span></span></code></pre></div>
<p>To remove biomolecules based on their presence in e_meta (such as
specifying proteins to remove from a pepData object), we implement the
filter as follows. This also removes the associated biomolecules from
e_data, as applicable.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">remove_prots</span> <span class="op">&lt;-</span> <span class="va">mypep</span><span class="op">$</span><span class="va">e_meta</span><span class="op">$</span><span class="va">RazorProtein</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="va">mycustomfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/custom_filter.html">custom_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, e_meta_remove <span class="op">=</span> <span class="va">remove_prots</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mycustomfilt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Custom Filter</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                        Filtered Remaining Total</span></span>
<span><span class="co">## SampleIDs (f_data)            0        24    24</span></span>
<span><span class="co">## Peptides (e_data)            59     21270 21329</span></span>
<span><span class="co">## RazorProteins (e_meta)       10      2972  2982</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mypep_custom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">mycustomfilt</span>, omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span><span class="op">)</span></span></code></pre></div>
<p>We can also utilize the custom filter and specify only those
biomolecules in either e_data or e_meta that we would like to keep. To
do this, use the arguments <code>e_data_keep</code> and
<code>e_meta_keep</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="sample-filters">Sample Filters<a class="anchor" aria-label="anchor" href="#sample-filters"></a>
</h2>
<p>The following filters remove entire samples from omicsData
objects.</p>
<div class="section level3">
<h3 id="rmd-pav-filter">rMd-PAV Filter<a class="anchor" aria-label="anchor" href="#rmd-pav-filter"></a>
</h3>
<p>The robust Mahalanobis distance squared values associated with the
peptide abundances vector (rMd-PAV) is a method which computes a rMd and
that can be mapped to a p-value and used to identify outlying samples
<span class="citation">(Matzke et al. 2011)</span>. Any identified
samples can be reviewed by the analyst to determine whether to remove
them from the dataset.</p>
<p>The metrics on which the log2 robust Mahalanobis distance is based
can be specified using the <code>metrics</code> argument of the
<code><a href="../reference/rmd_filter.html">rmd_filter()</a></code> function. Available metrics include:</p>
<ul>
<li><p>Median absolute deviation (MAD): specified as “MAD”</p></li>
<li><p>Correlation: specified as “Correlation”</p></li>
<li><p>Skewness: specified as “Skewness”</p></li>
<li><p>Kurtosis: specified as “Kurtosis”</p></li>
<li><p>Proportion Missing: specified as “Proportion_Missing”</p></li>
</ul>
<p>For pepData and proData objects, the default is to use all five of
the metrics. For metabData and lipidData objects, the default is to
exclude the proportion missing (as these data typically do not have much
missing data).</p>
<p>When plotting the RMD filter, we can specify a p-value threshold</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myrmdfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rmd_filter.html">rmd_filter</a></span><span class="op">(</span><span class="va">mypep_groups_log2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">myrmdfilt</span>, pvalue_threshold <span class="op">=</span> <span class="fl">0.0001</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/create%20rmd%20and%20plot-1.png" width="768"></p>
<p>If we want to look at individual samples, especially if there are
samples falling above the threshold line, we can do so using the
following. The colored points on each boxplot correspond to the sample
that was specified, so we can see where it falls relative to all of the
other samples.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">myrmdfilt</span>, sampleID <span class="op">=</span> <span class="st">"Sample_35_Phenotype2_A"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/rmd%20plot%20individ%20outliers-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="custom-filter-1">Custom Filter<a class="anchor" aria-label="anchor" href="#custom-filter-1"></a>
</h3>
<p>A custom filter can be used to remove entire samples. This is
particularly useful if only a subset of potential outliers identified by
the rMd filter are to be removed, or if quality control samples that
were run within the experiment are no longer needed for a particular
analysis step.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mycustomfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/custom_filter.html">custom_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">mypep_groups_log2</span>, f_data_remove <span class="op">=</span> <span class="st">"Sample_35_Phenotype2_A"</span><span class="op">)</span></span></code></pre></div>
<p>Similar to the usage of the custom filter for filtering biomolecules,
we could alternately use the <code>f_data_keep</code> argument to
specify which samples to retain in the dataset.</p>
</div>
</div>
<div class="section level2">
<h2 id="typical-order-of-filter-application">Typical Order of Filter Application<a class="anchor" aria-label="anchor" href="#typical-order-of-filter-application"></a>
</h2>
<p>This section applies to all but seqData objects.</p>
<p>A custom filter could be applied at any point, depending on the study
and dataset. For instance, if proteomics data and there are contaminant
proteins or reverse hits, a custom filter can be used first thing to
remove these. If the RMD filter identifies potential outliers but not
all are to be removed, a custom filter can be used to remove a subset of
samples.</p>
<ol style="list-style-type: decimal">
<li><p>Molecule filter</p></li>
<li><p>CV filter (if using–we often do not)</p></li>
<li><p>Proteomics filter (if proteomics data)</p></li>
<li><p>RMD filter</p></li>
<li><p>IMD-ANOVA filter</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="rnaseq-data-filters">RNAseq Data Filters<a class="anchor" aria-label="anchor" href="#rnaseq-data-filters"></a>
</h2>
<p>Due to RNAseq data being count data, the typical filters differ
compared to the other data types supported by <code>pmartR</code>.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq</span> <span class="op">&lt;-</span> <span class="va">rnaseq_object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myseq</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    </span></span>
<span><span class="co">## Class                       seqData</span></span>
<span><span class="co">## Unique SampleNames (f_data)      45</span></span>
<span><span class="co">## Unique Transcripts (e_data)   49568</span></span>
<span><span class="co">## Unique Genes (e_meta)         52222</span></span>
<span><span class="co">## Observed Zero-Counts         872181</span></span>
<span><span class="co">## Proportion Zeros              0.391</span></span></code></pre>
<p>The molecule filter and custom filter (either transcripts or samples)
are available for seqData objects, if useful for a particular dataset.
Not applicable for seqData objects are the CV filter, IMD-ANOVA filter,
and RMD filter.</p>
<div class="section level3">
<h3 id="total-count-filter">Total Count Filter<a class="anchor" aria-label="anchor" href="#total-count-filter"></a>
</h3>
<p>The total count filter is based on recommendations in edgeR
processing <span class="citation">(Robinson, McCarthy, and Smyth
2010)</span>, where the low-observed biomolecules are removed. The
default recommendation in edgeR is to require at least 10 total counts
observed across samples (i.e., if the sum of counts in a row of e_data
is &lt; 10, default edgeR filtering would remove this biomolecule).</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mytotcountfilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/total_count_filter.html">total_count_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">myseq</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mytotcountfilt</span>, min_count <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of Total Count Filter</span></span>
<span><span class="co">## ----------------------</span></span>
<span><span class="co">## Counts:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span><span class="co">##        1       14       79    15477     3194 15803184 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Number Filtered Biomolecules: 10299</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq_totcount</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">mytotcountfilt</span>, omicsData <span class="op">=</span> <span class="va">myseq</span>, min_count <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myseq_totcount</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    </span></span>
<span><span class="co">## Class                       seqData</span></span>
<span><span class="co">## Unique SampleNames (f_data)      45</span></span>
<span><span class="co">## Unique Transcripts (e_data)   39269</span></span>
<span><span class="co">## Unique Genes (e_meta)         42532</span></span>
<span><span class="co">## Observed Zero-Counts         440373</span></span>
<span><span class="co">## Proportion Zeros              0.249</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="rna-filter-total-library-size-filter">RNA Filter: Total Library Size Filter<a class="anchor" aria-label="anchor" href="#rna-filter-total-library-size-filter"></a>
</h3>
<p>The RNA filter removes samples based on either:</p>
<ul>
<li><p>library size (number of reads)</p></li>
<li><p>number/proportion of unique non-zero biomolecules per
sample</p></li>
</ul>
<p>This filter is particularly useful for identifying any samples that
contain lower than expected numbers of reads.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create RNA filter object</span></span>
<span><span class="va">myrnafilt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RNA_filter.html">RNA_filter</a></span><span class="op">(</span>omicsData <span class="op">=</span> <span class="va">myseq</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># filter by library size</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">myrnafilt</span>, plot_type <span class="op">=</span> <span class="st">"library"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">myrnafilt</span>, plot_type <span class="op">=</span> <span class="st">"library"</span>, size_library <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/unnamed-chunk-6-2.png" width="768"></p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myrnafilt</span>, size_library <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of RNA Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## Library size cut-off: 10000</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##         LibrarySize   NonZero ProportionNonZero</span></span>
<span><span class="co">## Min.       12208513 26359.000        0.53177453</span></span>
<span><span class="co">## 1st Qu.    14966913 28912.000        0.58327954</span></span>
<span><span class="co">## Median     16751973 29865.000        0.60250565</span></span>
<span><span class="co">## Mean       17048354 30186.200        0.60898564</span></span>
<span><span class="co">## 3rd Qu.    18517261 31299.000        0.63143560</span></span>
<span><span class="co">## Max.       24990997 35713.000        0.72048499</span></span>
<span><span class="co">## SD          3078177  1978.452        0.03991389</span></span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq_librarysize</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myrnafilt</span>, omicsData <span class="op">=</span> <span class="va">myseq</span>, size_library <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myseq_librarysize</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    </span></span>
<span><span class="co">## Class                       seqData</span></span>
<span><span class="co">## Unique SampleNames (f_data)      45</span></span>
<span><span class="co">## Unique Transcripts (e_data)   49568</span></span>
<span><span class="co">## Unique Genes (e_meta)         52222</span></span>
<span><span class="co">## Observed Zero-Counts         872181</span></span>
<span><span class="co">## Proportion Zeros              0.391</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter based on number or proportion of non-zero counts</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">myrnafilt</span>, plot_type <span class="op">=</span> <span class="st">"biomolecule"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Filter_Functionality_files/figure-html/unnamed-chunk-6-3.png" width="768"></p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myrnafilt</span>, min_nonzero <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of RNA Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## N Non-zero biomolecule cut-off: 5000</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##         LibrarySize   NonZero ProportionNonZero</span></span>
<span><span class="co">## Min.       12208513 26359.000        0.53177453</span></span>
<span><span class="co">## 1st Qu.    14966913 28912.000        0.58327954</span></span>
<span><span class="co">## Median     16751973 29865.000        0.60250565</span></span>
<span><span class="co">## Mean       17048354 30186.200        0.60898564</span></span>
<span><span class="co">## 3rd Qu.    18517261 31299.000        0.63143560</span></span>
<span><span class="co">## Max.       24990997 35713.000        0.72048499</span></span>
<span><span class="co">## SD          3078177  1978.452        0.03991389</span></span></code></pre>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myrnafilt</span>, min_nonzero <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Summary of RNA Filter </span></span>
<span><span class="co">## ----------------------------------</span></span>
<span><span class="co">## Proportion Non-zero biomolecule cut-off: 0.200000</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##         LibrarySize   NonZero ProportionNonZero</span></span>
<span><span class="co">## Min.       12208513 26359.000        0.53177453</span></span>
<span><span class="co">## 1st Qu.    14966913 28912.000        0.58327954</span></span>
<span><span class="co">## Median     16751973 29865.000        0.60250565</span></span>
<span><span class="co">## Mean       17048354 30186.200        0.60898564</span></span>
<span><span class="co">## 3rd Qu.    18517261 31299.000        0.63143560</span></span>
<span><span class="co">## Max.       24990997 35713.000        0.72048499</span></span>
<span><span class="co">## SD          3078177  1978.452        0.03991389</span></span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myseq_nonzero</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/applyFilt.html">applyFilt</a></span><span class="op">(</span>filter_object <span class="op">=</span> <span class="va">myrnafilt</span>, omicsData <span class="op">=</span> <span class="va">myseq</span>, min_nonzero <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">myseq_nonzero</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    </span></span>
<span><span class="co">## Class                       seqData</span></span>
<span><span class="co">## Unique SampleNames (f_data)      45</span></span>
<span><span class="co">## Unique Transcripts (e_data)   49568</span></span>
<span><span class="co">## Unique Genes (e_meta)         52222</span></span>
<span><span class="co">## Observed Zero-Counts         872181</span></span>
<span><span class="co">## Proportion Zeros              0.391</span></span></code></pre>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-ahmed1995pooling" class="csl-entry">
Ahmed, SE. 1995. <span>“A Pooling Methodology for Coefficient of
Variation.”</span> <em>Sankhy<span>ā</span>: The Indian Journal of
Statistics, Series B</em>, 57–75.
</div>
<div id="ref-matzke2011improved" class="csl-entry">
Matzke, Melissa M, Katrina M Waters, Thomas O Metz, Jon M Jacobs, Amy C
Sims, Ralph S Baric, Joel G Pounds, and Bobbie-Jo M Webb-Robertson.
2011. <span>“Improved Quality Control Processing of Peptide-Centric
LC-MS Proteomics Data.”</span> <em>Bioinformatics</em> 27 (20): 2866–72.
</div>
<div id="ref-edger" class="csl-entry">
Robinson, Mark D, Davis J McCarthy, and Gordon K Smyth. 2010.
<span>“edgeR: A Bioconductor Package for Differential Expression
Analysis of Digital Gene Expression Data.”</span>
<em>Bioinformatics</em> 26 (1): 139–40.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Lisa Bramer, Kelly Stratton.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
